scripts/backup-restore.sh

```diff
@@ -0,0 +1,401 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ -z "$CK8S_CONFIG_PATH" ]; then
+  log "ERROR: CK8S_CONFIG_PATH is not set."
+  log "Please set and export the variable before running the script."
+  exit 1
+fi
+
+usage() {
+  echo "Usage: $0 <action> <target>"
+  echo "Actions: backup | restore"
+  echo "Targets: opensearch | harbor | rclone | velero"
+  exit 1
+}
+
+# Sets a timestamp log before every message, also colors it purple to make it easier to distinct from rest of the text because why not
+PURPLE='\033[0;35m'
+NOCOLOR='\033[0m'
+log() {
+  echo -e "[${PURPLE}$(date +'%Y-%m-%d %H:%M:%S')${NOCOLOR}]: $1"
+}
+
+# allows the script to be run from anywhere
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
```

Comment:
Question: Why change cwd? Why not just save the dir and use absolute paths?

```suggestion
SCRIPT_DIR="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
```

Reply:
Only the subshell changes dir so it basically does the same thing no? but your version is way better since it actually works with symlinks as well I guess. Changing to that instead 👍